### DISCUSSION TITLE
sensor docker can't pull, x509 error

### DISCUSSION THREAD

**Discussion by rmcclint** [Security-Onion-Solutions/securityonion#8938] [Unsupported Versions]
Installing a sensor node, it won't pull any docker images. 

docker pull dc2-ids:5000/security-onion-solutions/so-suricata:2.3.180
Error response from daemon: Get "https://dc2-ids:5000/v2/": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0

tried to add this cert to CA certs, either I failed or it doesn't help. 
Maybe I have the incorrect version of some OS package? (Ubuntu)
Can I regenerate the cert "properly"? Can I add this environment variable, and where? 

thanks
Rick

**Comment by rmcclint** [comment]
workaround:  on the sensor run:
systemctl edit docker 

create a new file with 

[Service]
Environment="GODEBUG=x509ignoreCN=0"

save and exit, restart docker
pull then works.

**Comment by dougburks** [comment]
Are you using different versions of Ubuntu perhaps? From https://docs.securityonion.net/en/2.3/installation.html:
> If you’re not going to use our Security Onion ISO image and you’re building a distributed deployment, then we recommend keeping the base OS consistent across all nodes in the deployment. For example, there are known issues when trying to run Ubuntu 18.04 on the manager and Ubuntu 20.04 on the sensor.
